In [1]:
# Supress Warnings

import warnings
warnings.filterwarnings('ignore')

In [2]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

In [3]:
# Displaying float upto 2 decimal points

pd.set_option('display.float_format', '{:.2f}'.format)

## Task 1: Loading and cleaning Data

-  ### Subtask 1.1: Import and read companies data.

Import and read the companies data. Store it in a dataframe called `companies`.

In [4]:
companies = pd.read_csv('companies.txt', sep='\t', encoding='ISO-8859-1')

#Inspecting companies dataframe

print(companies.shape)

print(companies.info())

companies.describe()

(66368, 10)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB
None


permalink   name                homepage_url  \
count                        66368  66367                       61310   
unique                       66368  66102                       61191   
top     /Organization/Comic-Rocket  Blink  http://www.askforoffer.com   
freq                             1      4                           5   

       category_list     status country_code state_code       region  \
count          63220      66368        59410      57821        58338   
unique         27296          4          137        311         1092   
top         Software  operating          USA         CA  SF Bay Area   
freq            3995      53034        37601      12900         8804   

                 city  founded_at  
count           58340       51147  
unique           5111        3978  
top     San Francisco  01-01-2012  
freq             3526        2730

In [5]:
# On inspecting the data from the above info and describe we can see the permalink column
# is having same number of rows as the number of elements in the dataframe. Hence it can 
# be used as key.This finding can be used later for drawing further inferences. 

companies.head(10) 


permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   
5     /Organization/01Games-Technology      01Games Technology   
6  /Organization/0Ndine-Biomedical-Inc  Ondine Biomedical Inc.   
7                 /Organization/0Xdata                  H2O.ai   
8                      /Organization/1                One Inc.   
9            /Organization/1-2-3-Listo             1,2,3 Listo   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   
5             http://www.01games.hk/   
6               http://ondinebio.com   
7                     http://h2o.ai/   
8                 http://whatis1.com   
9            http://www.123listo.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   
5                                              Games  operating          HKG   
6                                      Biotechnology  operating          CAN   
7                                          Analytics  operating          USA   
8                                             Mobile  operating          USA   
9                                         E-Commerce  operating          CHL   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010  
5        NaN              Hong Kong      Hong Kong         NaN  
6         BC              Vancouver      Vancouver  01-01-1997  
7         CA            SF Bay Area  Mountain View  01-01-2011  
8         CA            SF Bay Area  San Francisco  01-08-2011  
9         12               Santiago     Las Condes  01-01-2012

In [6]:
# The above inspection shows that there are still some special charecters included. Hence,
# for non-erroneous results , the encoding problem needs to be handled. 

companies['permalink'] = companies['permalink'].str.encode('utf-8').str.decode('ascii', 'ignore')

companies.head(10)

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   
5     /Organization/01Games-Technology      01Games Technology   
6  /Organization/0Ndine-Biomedical-Inc  Ondine Biomedical Inc.   
7                 /Organization/0Xdata                  H2O.ai   
8                      /Organization/1                One Inc.   
9            /Organization/1-2-3-Listo             1,2,3 Listo   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   
5             http://www.01games.hk/   
6               http://ondinebio.com   
7                     http://h2o.ai/   
8                 http://whatis1.com   
9            http://www.123listo.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   
5                                              Games  operating          HKG   
6                                      Biotechnology  operating          CAN   
7                                          Analytics  operating          USA   
8                                             Mobile  operating          USA   
9                                         E-Commerce  operating          CHL   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010  
5        NaN              Hong Kong      Hong Kong         NaN  
6         BC              Vancouver      Vancouver  01-01-1997  
7         CA            SF Bay Area  Mountain View  01-01-2011  
8         CA            SF Bay Area  San Francisco  01-08-2011  
9         12               Santiago     Las Condes  01-01-2012



-  ### Subtask 1.2: Import and read rounds2 data.

Import and read the round_2 data. Store it in a dataframe called `rounds2`.

In [7]:
rounds2 = pd.read_csv('rounds2.csv', sep=',', encoding='ISO-8859-1')

# Inspecting rounds2 dataframe

print(rounds2.shape)

print(rounds2.info())

rounds2.describe()


(114949, 6)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null object
raised_amount_usd          94959 non-null float64
dtypes: float64(1), object(5)
memory usage: 5.3+ MB
None


raised_amount_usd
count           94959.00
mean         10426869.33
std         114821247.98
min                 0.00
25%            322500.00
50%           1680511.00
75%           7000000.00
max       21271935000.00

In [8]:
rounds2.head(10)

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   
5       /ORGANIZATION/004-TECHNOLOGIES   
6     /organization/01games-technology   
7  /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8  /organization/0ndine-biomedical-inc   
9                 /ORGANIZATION/0XDATA   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015        10000000.00  
1                  A  14-10-2014                nan  
2                NaN  01-03-2014          700000.00  
3                  B  30-01-2014         3406878.00  
4                  A  19-03-2008         2000000.00  
5                NaN  24-07-2014                nan  
6                NaN  01-07-2014           41250.00  
7                NaN  11-09-2009           43360.00  
8                NaN  21-12-2009          719491.00  
9                NaN  22-05-2013         3000000.00

In [9]:
# The above inspection shows that there are still some special charecters included. 
# Hence, for non-erroneous results, the encoding problem needs to be handled. 

rounds2['company_permalink'] = rounds2['company_permalink'].str.encode('utf-8').str.decode('ascii', 'ignore')

rounds2.head(10)

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   
5       /ORGANIZATION/004-TECHNOLOGIES   
6     /organization/01games-technology   
7  /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8  /organization/0ndine-biomedical-inc   
9                 /ORGANIZATION/0XDATA   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015        10000000.00  
1                  A  14-10-2014                nan  
2                NaN  01-03-2014          700000.00  
3                  B  30-01-2014         3406878.00  
4                  A  19-03-2008         2000000.00  
5                NaN  24-07-2014                nan  
6                NaN  01-07-2014           41250.00  
7                NaN  11-09-2009           43360.00  
8                NaN  21-12-2009          719491.00  
9                NaN  22-05-2013         3000000.00

In [10]:
# From the above 2 dataframes we can see that permalink and company permalink can be 
# used for joining. However, first it needs to be formatted properly.

rounds2['company_permalink'] = rounds2['company_permalink'].str.lower()

companies['permalink'] = companies['permalink'].str.lower()

-  ### Subtask 1.3: Print unique records present in both the dataframes and identify the key column for companies.

Print count of unique records in `companies` and `rounds2`.

In [11]:
# Finding the key column. Both row count and unique values should be same.

print(companies.shape[0])

print(companies['permalink'].nunique())

print(companies['permalink'].is_unique) #if true than can be used as unique key column. 
# From here we conculde this as unique key column.

print(rounds2['company_permalink'].nunique())

66368
66368
True
66368


-  ### Subtask 1.4: Find if there are companies in the rounds2 file which are not present in companies.

Get the difference of `rounds2` and `companies`.

In [12]:
# From above we can observe that both dataframes have same count of records. 
# But these might be different. Hence we need to check if all companies present 
# in 'rounds2' are in 'companies'or not.


# To compare the values of permalink column of 'companies' with the 
# company_permalink column of 'rounds2'
unique_comp_companies = set(companies['permalink']) 
unique_comp_rounds2 = set(rounds2['company_permalink']) 


# printing the total number of elements after dropping the duplicates.
print(len(unique_comp_companies))
print(len(unique_comp_rounds2))


print(len(unique_comp_rounds2.difference(unique_comp_companies))) #if this is 0 then it means 
# that there is no company in the rounds2 file which is not present in companies

66368
66368
0


-  ### Subtask 1.5: Create master_frame by joining companies and rounds2.

Joining `companies` and `rounds2` dataframe to create `master_frame`.

In [13]:
master_frame = pd.merge(rounds2, companies, left_on=['company_permalink'], right_on =['permalink'] , how='inner')

# Inspecting master_frame

print(master_frame.shape)

print(companies.info())

master_frame.head(10)

(114949, 16)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       51147 non-null object
dtypes: object(10)
memory usage: 5.1+ MB
None


company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   
5       /organization/004-technologies   
6     /organization/01games-technology   
7  /organization/0ndine-biomedical-inc   
8  /organization/0ndine-biomedical-inc   
9                 /organization/0xdata   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015        10000000.00   
1                  A  14-10-2014                nan   
2                NaN  01-03-2014          700000.00   
3                  B  30-01-2014         3406878.00   
4                  A  19-03-2008         2000000.00   
5                NaN  24-07-2014                nan   
6                NaN  01-07-2014           41250.00   
7                NaN  11-09-2009           43360.00   
8                NaN  21-12-2009          719491.00   
9                NaN  22-05-2013         3000000.00   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   
5       /organization/004-technologies        004 Technologies   
6     /organization/01games-technology      01Games Technology   
7  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
8  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
9                 /organization/0xdata                  H2O.ai   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2             http://www.qounter.com   
3                http://oneofthem.jp   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   
6             http://www.01games.hk/   
7               http://ondinebio.com   
8               http://ondinebio.com   
9                     http://h2o.ai/   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2  Application Platforms|Real Time|Social Network...  operating          USA   
3                                  Apps|Games|Mobile  operating          NaN   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   
6                                              Games  operating          HKG   
7                                      Biotechnology  operating          CAN   
8                                      Biotechnology  operating          CAN   
9                                          Analytics  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumba

In [14]:
# To find the total number of null values for each column in master_frame.

print(master_frame.isnull().sum())


company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83809
funded_at                      0
raised_amount_usd          19990
permalink                      0
name                           1
homepage_url                6134
category_list               3410
status                         0
country_code                8678
state_code                 10946
region                     10167
city                       10164
founded_at                 20521
dtype: int64


In [15]:
# Getting the column-wise null percentage in master_frame

print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2)) 


# The raised_amount_usd column has 17.4% of the records missing. 
# The country_code column has 7.55% of the records missing.
# The category_list column has 2.97% of the records missing.


company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        72.91
funded_at                  0.00
raised_amount_usd         17.39
permalink                  0.00
name                       0.00
homepage_url               5.34
category_list              2.97
status                     0.00
country_code               7.55
state_code                 9.52
region                     8.84
city                       8.84
founded_at                17.85
dtype: float64


## Task 2: Finding the Average values of investment for funding types

-  ### Subtask 2.1: Cleaning master_frame.

Cleaning master_frame by removing rows with missing vales in raised_amount_usd column as raised_amount_usd column is having 17.39% missing values

In [16]:
# We have observed earlier that raised_amount_usd column has 17.4% of the records missing. 
# Hence, removing rows with missing values in raised_amount_usd column.

master_frame = master_frame[~np.isnan(master_frame['raised_amount_usd'])]   


# Getting the column-wise null percentage
round(100*(master_frame.isnull().sum()/len(master_frame.index)),2) 

company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        70.34
funded_at                  0.00
raised_amount_usd          0.00
permalink                  0.00
name                       0.00
homepage_url               4.56
category_list              1.10
status                     0.00
country_code               6.16
state_code                 8.01
region                     7.42
city                       7.42
founded_at                16.81
dtype: float64

-  ### Subtask 2.2: Find the average amount raised based on funding round type.

Using master_frame to groupby on funding_round_type and calculating average raised fund amount.

In [17]:
# Finding average investment for each funding round type
avg_invest_by_funding_round = pd.DataFrame(master_frame.groupby('funding_round_type').raised_amount_usd.mean())

# Finding median investment for each funding round type
invest_by_funding_round_median = pd.DataFrame(master_frame.groupby('funding_round_type').raised_amount_usd.median())

print(avg_invest_by_funding_round)

print(invest_by_funding_round_median)


                       raised_amount_usd
funding_round_type                      
angel                          958694.47
convertible_note              1453438.54
debt_financing               17043526.02
equity_crowdfunding            538368.21
grant                         4300576.34
non_equity_assistance          411203.05
post_ipo_debt               168704571.82
post_ipo_equity              82182493.87
private_equity               73308593.03
product_crowdfunding          1363131.07
secondary_market             79649630.10
seed                           719818.00
undisclosed                  19242370.23
venture                      11748949.13
                       raised_amount_usd
funding_round_type                      
angel                          400000.00
convertible_note               272000.00
debt_financing                1100000.00
equity_crowdfunding            100000.00
grant                          201684.00
non_equity_assistance           60000.00
post_ipo_debt   

-  ### Subtask 2.3: Find the funding round type where average investment per round is between 5 to 15 million USD per investment round.

Using avg_invest_by_funding_round to find funding round type where average investment per round is between 5 to 15 million USD per investment round.

In [18]:
avg_invest_by_funding_round.loc[(avg_invest_by_funding_round.raised_amount_usd>=5000000) & (avg_invest_by_funding_round.raised_amount_usd<=15000000)]

# Range specified is 5 to 15 million USD. So from above we can take that only venture funding round 
# is suitable for Spark funds.
# Also, we can see the median value for venture also lies between 5 to 15 million USD.

raised_amount_usd
funding_round_type                   
venture                   11748949.13

## Task 3: Finding the Countries where most investment are occuring for venture funding type

-  ### Subtask 3.1: Cleaning and Filtering master_frame for venture funding type



In [19]:
# We have observed earlier that country_code column has 7.55% missing values.
# Hence, Removing rows with missing values in country_code column.
master_frame = master_frame[~master_frame['country_code'].isnull()]   


# Getting the column-wise null percentage
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))

company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        69.77
funded_at                  0.00
raised_amount_usd          0.00
permalink                  0.00
name                       0.00
homepage_url               3.79
category_list              0.65
status                     0.00
country_code               0.00
state_code                 1.97
region                     1.34
city                       1.34
founded_at                15.33
dtype: float64


In [20]:
# Filtering master_frame to get those investments only where funding round type is venture
venture_funding = master_frame.loc[(master_frame['funding_round_type']=='venture')]


# Grouping the venture_funding dataframe using the country_code column and 
# getting total raised_amount_usd per country
venture_funding_by_country = venture_funding.groupby('country_code').raised_amount_usd.sum()


# Taking top nine countries which have received the highest total funding 
# (across ALL sectors for the venture investment type)
top9 = pd.DataFrame(venture_funding_by_country).sort_values('raised_amount_usd', ascending=False).head(9)
top9

raised_amount_usd
country_code                   
USA             422510842796.00
CHN              39835418773.00
GBR              20245627416.00
IND              14391858718.00
CAN               9583332317.00
FRA               7259536732.00
ISR               6907514579.00
DEU               6346959822.00
JPN               3363676611.00

In [21]:
# We can inspect the codes in the above list manually with data available over internet
# and map them to the country corresponding to it.

# We can observe that :-
# USA is at the top and is an English Speaking Country. 
# CHN is second but is not an English Speaking Country based on the list above. 
# GBR will be taken as Second as it is an English Speaking Country.
# IND will be taken as Third as it is an English Speaking Country.

# Finding the top 3 english speaking countries.
top3_english = top9.loc[['USA','GBR','IND']]
top3_english

raised_amount_usd
country_code                   
USA             422510842796.00
GBR              20245627416.00
IND              14391858718.00

## Task 4: Sector Analysis

-  ### Subtask 4.1: Cleaning and Filtering master_frame
Create new columns primary_sector and other_sectors from category_list



In [22]:
# We observed earlier that category_list column has 2.97% values missing 

# Removing rows where category_list column has missing values
master_frame = master_frame[~master_frame['category_list'].isnull()]   

# Getting the column-wise null percentage
print(round(100*(master_frame.isnull().sum()/len(master_frame.index)),2))


company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        69.75
funded_at                  0.00
raised_amount_usd          0.00
permalink                  0.00
name                       0.00
homepage_url               3.67
category_list              0.00
status                     0.00
country_code               0.00
state_code                 1.96
region                     1.34
city                       1.33
founded_at                15.10
dtype: float64


In [23]:
# extracting the primary sector from category_list

master_frame[['primary_sector', 'other_sectors']]= master_frame['category_list'].str.split('|', n=1, expand=True)

master_frame.head(10)

company_permalink  \
0                   /organization/-fame   
2                /organization/-qounter   
4                 /organization/0-6-com   
6      /organization/01games-technology   
7   /organization/0ndine-biomedical-inc   
8   /organization/0ndine-biomedical-inc   
9                  /organization/0xdata   
10                 /organization/0xdata   
11                 /organization/0xdata   
12                 /organization/0xdata   

                            funding_round_permalink funding_round_type  \
0   /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2   /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4   /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6   /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7   /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8   /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9   /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   
10  /funding-round/3bb2ee4a2d89251a10aaa735b1180e44            venture   
11  /funding-round/ae2a174c06517c2394aed45006322a7e            venture   
12  /funding-round/e1cfcbe1bdf4c70277c5f29a3482f24e            venture   

   funding_round_code   funded_at  raised_amount_usd  \
0                   B  05-01-2015        10000000.00   
2                 NaN  01-03-2014          700000.00   
4                   A  19-03-2008         2000000.00   
6                 NaN  01-07-2014           41250.00   
7                 NaN  11-09-2009           43360.00   
8                 NaN  21-12-2009          719491.00   
9                 NaN  22-05-2013         3000000.00   
10                  B  09-11-2015        20000000.00   
11                NaN  03-01-2013         1700000.00   
12                  A  19-07-2014         8900000.00   

                              permalink                    name  \
0                   /organization/-fame                   #fame   
2                /organization/-qounter                :Qounter   
4                 /organization/0-6-com                 0-6.com   
6      /organization/01games-technology      01Games Technology   
7   /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
8   /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
9                  /organization/0xdata                  H2O.ai   
10                 /organization/0xdata                  H2O.ai   
11                 /organization/0xdata                  H2O.ai   
12                 /organization/0xdata                  H2O.ai   

              homepage_url                                      category_list  \
0       http://livfame.com                                              Media   
2   http://www.qounter.com  Application Platforms|Real Time|Social Network...   
4       http://www.0-6.com                                        Curated Web   
6   http://www.01games.hk/                                              Games   
7     http://ondinebio.com                                      Biotechnology   
8     http://ondinebio.com                                      Biotechnology   
9           http://h2o.ai/                                          Analytics   
10          http://h2o.ai/                                          Analytics   
11          http://h2o.ai/                                          Analytics   
12          http://h2o.ai/                                          Analytics   

       status country_code state_code       region           city  founded_at  \
0   operating          IND         16       Mumbai         Mumbai         NaN   
2   operating          USA         DE   DE - Other  Delaware City  04-09-2014   
4   operating          CHN         22      Beijing        Beijing  01-01-2007   
6   operating          HKG        NaN    Hong Kong      Hong Kong         NaN   
7   operating          CAN         BC    Vancouver      Vancouver  01-01-1997   


-  ### Subtask 4.2: Reading mapping.csv
Read mapping.csv and transform the data as required.

In [24]:
# importing the csv file into sector_mapping dataframe

sector_mapping = pd.read_csv('mapping.csv', encoding = "ISO-8859-1")  

sector_mapping.head(15)

category_list  Automotive & Sports  Blanks  \
0                     NaN                    0       1   
1                      3D                    0       0   
2             3D Printing                    0       0   
3           3D Technology                    0       0   
4              Accounting                    0       0   
5        Active Lifestyle                    0       0   
6            Ad Targeting                    0       0   
7      Advanced Materials                    0       0   
8        Adventure Travel                    1       0   
9             Advertising                    0       0   
10  Advertising Exchanges                    0       0   
11   Advertising Networks                    0       0   
12  Advertising Platforms                    0       0   
13                 Advice                    0       0   
14              Aerospace                    1       0   

    Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
0                            0              0       0              0   
1                            0              0       0              1   
2                            0              0       0              1   
3                            0              0       0              1   
4                            0              0       0              0   
5                            0              0       1              0   
6                            0              0       0              0   
7                            0              0       0              1   
8                            0              0       0              0   
9                            0              0       0              0   
10                           0              0       0              0   
11                           0              0       0              0   
12                           0              0       0              0   
13                           0              0       0              0   
14                           0              0       0              0   

    News, Search and Messaging  Others  \
0                            0       0   
1                            0       0   
2                            0       0   
3                            0       0   
4                            0       0   
5                            0       0   
6                            0       0   
7                            0       0   
8                            0       0   
9                            0       0   
10                           0       0   
11                           0       0   
12                           0       0   
13                           0       1   
14                           0       0   

    Social, Finance, Analytics, Advertising  
0                                         0  
1                                         0  
2                                         0  
3                                         0  
4                                         1  
5                                         0  
6                                         1  
7                                         0  
8                                         0  
9                                         1  
10                                        1  
11                                        1  
12                                        1  
13                                        0  
14                                        0

In [25]:
# Getting the column-wise null percentage in sector_mapping dataframe

print(round(100*(sector_mapping.isnull().sum()/len(sector_mapping.index)),2))

category_list                             0.15
Automotive & Sports                       0.00
Blanks                                    0.00
Cleantech / Semiconductors                0.00
Entertainment                             0.00
Health                                    0.00
Manufacturing                             0.00
News, Search and Messaging                0.00
Others                                    0.00
Social, Finance, Analytics, Advertising   0.00
dtype: float64


In [26]:
# As category_list has 0.15 % missing values,
# Hence, removing rows with missing values in category_list column.
sector_mapping = sector_mapping[~sector_mapping['category_list'].isnull()]   


# Getting the column-wise null percentage
print(round(100*(sector_mapping.isnull().sum()/len(sector_mapping.index)),2) )

category_list                             0.00
Automotive & Sports                       0.00
Blanks                                    0.00
Cleantech / Semiconductors                0.00
Entertainment                             0.00
Health                                    0.00
Manufacturing                             0.00
News, Search and Messaging                0.00
Others                                    0.00
Social, Finance, Analytics, Advertising   0.00
dtype: float64


In [27]:
# on looking into csv, we need to replace '0's with 'na' in the category_list.
# e.g. A0lytics should become Analytics
sector_mapping['category_list'] = sector_mapping['category_list'].str.replace('0','na')

# We can see that there are a total of 8 primary_sectors including others.

In [28]:
# Transforming the sector_mapping to get only the category_list and its corresponding main sector

# In sector_mapping, finding the main sector(where column value=1) for each category_list 
# and putting in main_sector column
sector_mapping['main_sector'] = sector_mapping[sector_mapping.columns.difference(['category_list'])].idxmax(axis=1)

sector_mapping.head(10)

category_list  Automotive & Sports  Blanks  \
1                      3D                    0       0   
2             3D Printing                    0       0   
3           3D Technology                    0       0   
4              Accounting                    0       0   
5        Active Lifestyle                    0       0   
6            Ad Targeting                    0       0   
7      Advanced Materials                    0       0   
8        Adventure Travel                    1       0   
9             Advertising                    0       0   
10  Advertising Exchanges                    0       0   

    Cleantech / Semiconductors  Entertainment  Health  Manufacturing  \
1                            0              0       0              1   
2                            0              0       0              1   
3                            0              0       0              1   
4                            0              0       0              0   
5                            0              0       1              0   
6                            0              0       0              0   
7                            0              0       0              1   
8                            0              0       0              0   
9                            0              0       0              0   
10                           0              0       0              0   

    News, Search and Messaging  Others  \
1                            0       0   
2                            0       0   
3                            0       0   
4                            0       0   
5                            0       0   
6                            0       0   
7                            0       0   
8                            0       0   
9                            0       0   
10                           0       0   

    Social, Finance, Analytics, Advertising  \
1                                         0   
2                                         0   
3                                         0   
4                                         1   
5                                         0   
6                                         1   
7                                         0   
8                                         0   
9                                         1   
10                                        1   

                                main_sector  
1                             Manufacturing  
2                             Manufacturing  
3                             Manufacturing  
4   Social, Finance, Analytics, Advertising  
5                                    Health  
6   Social, Finance, Analytics, Advertising  
7                             Manufacturing  
8                       Automotive & Sports  
9   Social, Finance, Analytics, Advertising  
10  Social, Finance, Analytics, Advertising

In [29]:
# formatting the sector_mapping further to retain only category_list and its main_sector

sector_mapping = sector_mapping[['category_list','main_sector']]

sector_mapping.head(10)


category_list                              main_sector
1                      3D                            Manufacturing
2             3D Printing                            Manufacturing
3           3D Technology                            Manufacturing
4              Accounting  Social, Finance, Analytics, Advertising
5        Active Lifestyle                                   Health
6            Ad Targeting  Social, Finance, Analytics, Advertising
7      Advanced Materials                            Manufacturing
8        Adventure Travel                      Automotive & Sports
9             Advertising  Social, Finance, Analytics, Advertising
10  Advertising Exchanges  Social, Finance, Analytics, Advertising

-  ### Subtask 4.3: Joining master_frame and sector_mapping.
Joining master_frame and sector_mapping and doing the necessary formatting.

In [30]:
master_frame['primary_sector'] = master_frame['primary_sector'].str.lower()

sector_mapping['category_list'] = sector_mapping['category_list'].str.lower()


# Merging masterframe with sector_mapping so that each primary sector gets mapped to its main sector
merged_frame = pd.merge(master_frame, sector_mapping, left_on=['primary_sector'], right_on =['category_list'] , how='left', indicator='True')


# Difference is due to the fact that we did have some primary_sectors in master frame 
# for which no mapped category_list were present in mapping file.
difference_frame = merged_frame[merged_frame['True']!='both']
difference_frame

# From this we can see that there are some primary sectors which are not in the mapping file
# Hence, We can ignore these by merging these dataframes by INNER JOIN. 

company_permalink  \
1995                             /organization/adora-inc   
4947   /organization/apollo-commercial-real-estate-fi...   
6358                        /organization/asia-translate   
9758                              /organization/bioserie   
9759                              /organization/bioserie   
9760                              /organization/bioserie   
10772                            /organization/blurr-llc   
13745                          /organization/cartpay-co-   
13746                          /organization/cartpay-co-   
13751                             /organization/carusele   
13752                             /organization/carusele   
15903              /organization/clean-vehicle-solutions   
16218                          /organization/clientscape   
17155                            /organization/cokonnect   
17156                            /organization/cokonnect   
17529      /organization/compact-power-equipment-centers   
17993                      /organization/consensus-point   
17994                      /organization/consensus-point   
17995                      /organization/consensus-point   
17996                      /organization/consensus-point   
17997                      /organization/consensus-point   
17998                      /organization/consensus-point   
18098                      /organization/content-circles   
22354              /organization/dolores-speech-products   
24709      /organization/endavo-media-and-communications   
25706                         /organization/esports-hero   
27532                        /organization/fia-formula-e   
28076                           /organization/five-below   
28280                             /organization/flexport   
28281                             /organization/flexport   
...                                                  ...   
58939                          /organization/playfitness   
59302                                /organization/podio   
59303                                /organization/podio   
59304                                /organization/podio   
59378               /organization/polaris-design-systems   
59611                              /organization/pornhub   
60066          /organization/prediki-prediction-services   
61263  /organization/psychological-service-you-can-ha...   
63748          /organization/regency-energy-partners-llc   
64731                         /organization/rezonver-llc   
64732                         /organization/rezonver-llc   
64733                         /organization/rezonver-llc   
67761                      /organization/sense-labs-inc-   
68301           /organization/shantiniketan-incorporated   
68302           /organization/shantiniketan-incorporated   
69512                              /organization/sincuru   
69785                     /organization/skillgravity-com   
71679                  /organization/sos-by-bernard-buie   
77334                     /organization/third-eye-health   
81050                /organization/us-emergency-registry   
81761                             /organization/velocomp   
81779                           /organization/ven-racing   
83469                               /organization/vnatek   
85062                          /organization/whirly-golf   
85942                          /organization/worldwinger   
85943                          /organization/worldwinger   
86540                               /organization/xormis   
86894      /organization/yellowstone-compact-commodities   
86895      /organization/yellowstone-compact-commodities   
87076                              /organization/yoolink   

                               funding_round_permalink    funding_round_type  \
1995   /funding-round/086235eb44845d9bfdde633ce4c468b1                  seed   
4947   /funding-round/c2cb230111ce9a4acf6fd87b97b8741e        debt_financing   
6358   /funding-round/0e445561d485e211ccc409be15562dd9               venture  

In [31]:
# Merging master_frame and sector_mapping by INNER JOIN
merged_frame = pd.merge(master_frame, sector_mapping, left_on=['primary_sector'], right_on =['category_list'] , how='inner')

# dropping few unnecessary columns that came after merge
merged_frame = merged_frame.drop(['category_list_y','other_sectors'], axis=1)

# Getting the column-wise null percentage
print(round(100*(merged_frame.isnull().sum()/len(merged_frame.index)),2) )


company_permalink          0.00
funding_round_permalink    0.00
funding_round_type         0.00
funding_round_code        69.73
funded_at                  0.00
raised_amount_usd          0.00
permalink                  0.00
name                       0.00
homepage_url               3.67
category_list_x            0.00
status                     0.00
country_code               0.00
state_code                 1.95
region                     1.34
city                       1.33
founded_at                15.10
primary_sector             0.00
main_sector                0.00
dtype: float64


In [32]:
merged_frame.head(10)

company_permalink  \
0                /organization/-fame   
1                /organization/90min   
2                /organization/90min   
3                /organization/90min   
4       /organization/a-dance-for-me   
5         /organization/akira-mobile   
6        /organization/all-day-media   
7      /organization/all-def-digital   
8  /organization/america-s-real-deal   
9      /organization/american-gnuity   

                           funding_round_permalink   funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638              venture   
1  /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6              venture   
2  /funding-round/bd626ed022f5c66574b1afe234f3c90d              venture   
3  /funding-round/fd4b15e8c97ee2ffc0acccdbe1a98810              venture   
4  /funding-round/9ab9dbd17bf010c79d8415b2c22be6fa  equity_crowdfunding   
5  /funding-round/bfb170aea580e381e5b1810c87855c9e                 seed   
6  /funding-round/2760c0426a124b84c540bd4fd2dfe6e5                 seed   
7  /funding-round/452a2342fe720285c3b92e9bd927d9ba              venture   
8  /funding-round/5fbb191b97ca9789a7196de703983240  equity_crowdfunding   
9  /funding-round/07cc3eb3afd8ef5812e2b62b254b2040  equity_crowdfunding   

  funding_round_code   funded_at  raised_amount_usd  \
0                  B  05-01-2015        10000000.00   
1                NaN  06-10-2015        15000000.00   
2                NaN  07-05-2013         5800000.00   
3                NaN  26-03-2014        18000000.00   
4                NaN  26-03-2014         1090000.00   
5                NaN  01-08-2012           18410.00   
6                NaN  16-12-2014         2000000.00   
7                  A  06-08-2014         5000000.00   
8                NaN  19-08-2011          670000.00   
9                NaN  19-08-2011          670000.00   

                           permalink                 name  \
0                /organization/-fame                #fame   
1                /organization/90min                90min   
2                /organization/90min                90min   
3                /organization/90min                90min   
4       /organization/a-dance-for-me       A Dance for Me   
5         /organization/akira-mobile         Akira Mobile   
6        /organization/all-day-media        ALL DAY MEDIA   
7      /organization/all-def-digital      All Def Digital   
8  /organization/america-s-real-deal  America's Real Deal   
9      /organization/american-gnuity      American Gnuity   

                   homepage_url                      category_list_x  \
0            http://livfame.com                                Media   
1          http://www.90min.com  Media|News|Publishing|Soccer|Sports   
2          http://www.90min.com  Media|News|Publishing|Soccer|Sports   
3          http://www.90min.com  Media|News|Publishing|Soccer|Sports   
4   http://www.adanceforme.com/       Media|News|Photo Sharing|Video   
5        http://www.akira.lt/en  Media|Mobile|SMS|Telecommunications   
6            http://allday.com/                   Media|Social Media   
7      http://alldefdigital.com                                Media   
8  http://americasrealdeal.com/                           Media|News   
9     http://AmericanGnuity.com                           Media|News   

      status country_code state_code          region         city  founded_at  \
0  operating          IND         16          Mumbai       Mumbai         NaN   
1  operating          GBR         H9          London       London  01-01-2011   
2  operating          GBR         H9          London       London  01-01-2011   
3  operating          GBR         H9          London       London  01-01-2011   
4  operating          USA         MT        Missoula     Missoula  31-07-2011   
5  operating          LTU        NaN             NaN          NaN         NaN   
6  operating          USA         CA     Los Angeles  Los Angeles  01-01-2013   
7  operating          USA        

## Task 5: Sector Analysis 

-  ### Subtask 5.1: Finding out the most heavily invested main sectors in each of the three countries (for funding type FT and investments range of 5-15 M USD)
First we would create 1 common dataframe for all 3 countries then derive D1, D2, D3 from it.


In [33]:
# Filtering merged_frame for top3 countries
D = merged_frame.loc[merged_frame['country_code'].isin(['USA', 'GBR', 'IND'])]

# Filtering D for funding_round_type = venture and raised_amount_usd between 5-15 million
D = merged_frame.loc[((merged_frame['funding_round_type']=='venture') & (merged_frame.raised_amount_usd>=5000000) & (merged_frame.raised_amount_usd<=15000000))]

#The total number (or count) of investments for each main sector in a separate column based on country
D['total_number_of_investment'] = D.groupby(['country_code','main_sector'])['main_sector'].transform('count')


#The total amount invested in each main sector in a separate column
D['total_amount_invested'] = D.groupby(['country_code','main_sector'])['raised_amount_usd'].transform('sum')

# Creating specific dataframes for each country
D1 = D.loc[(D['country_code']=='USA')]
D2 = D.loc[(D['country_code']=='GBR')]
D3 = D.loc[(D['country_code']=='IND')]


-  ### Doing for country_code == USA  
      -  ### (United States)

In [34]:
# inspecting the created dataframe for USA

D1.info()

D1.head(10)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 12063 entries, 7 to 88439
Data columns (total 20 columns):
company_permalink             12063 non-null object
funding_round_permalink       12063 non-null object
funding_round_type            12063 non-null object
funding_round_code            8618 non-null object
funded_at                     12063 non-null object
raised_amount_usd             12063 non-null float64
permalink                     12063 non-null object
name                          12063 non-null object
homepage_url                  11495 non-null object
category_list_x               12063 non-null object
status                        12063 non-null object
country_code                  12063 non-null object
state_code                    12061 non-null object
region                        12054 non-null object
city                          12054 non-null object
founded_at                    10390 non-null object
primary_sector                12063 non-null object
main_se

company_permalink  \
7    /organization/all-def-digital   
31        /organization/chefs-feed   
61    /organization/huffingtonpost   
62    /organization/huffingtonpost   
85         /organization/matchmine   
88         /organization/mediabong   
97   /organization/newscorporation   
102          /organization/nokeena   
104          /organization/nokeena   
113           /organization/plumtv   

                             funding_round_permalink funding_round_type  \
7    /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
31   /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
61   /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
62   /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
85   /funding-round/41ac526630da57ad6eb9d02431b17657            venture   
88   /funding-round/9282890ca87072025dc1807f400acee6            venture   
97   /funding-round/8f6d7c4592e43e91e8688ba342bffcb7            venture   
102  /funding-round/9225f2db6b1b74892d5de6a8744b94ea            venture   
104  /funding-round/fb7a617ed6d1a3203024f0f111417bfb            venture   
113  /funding-round/e5109c28c1b4899b068cfa552850c424            venture   

    funding_round_code   funded_at  raised_amount_usd  \
7                    A  06-08-2014         5000000.00   
31                   A  26-02-2015         5000000.00   
61                   A  01-08-2006         5000000.00   
62                   B  01-09-2007         5000000.00   
85                   A  01-09-2007        10000000.00   
88                   B  13-05-2015         5000000.00   
97                 NaN  08-01-2010        12500000.00   
102                  A  08-08-2008         9400000.00   
104                  B  25-06-2009         6500000.00   
113                  B  01-09-2009         5200000.00   

                         permalink                 name  \
7    /organization/all-def-digital      All Def Digital   
31        /organization/chefs-feed            ChefsFeed   
61    /organization/huffingtonpost  The Huffington Post   
62    /organization/huffingtonpost  The Huffington Post   
85         /organization/matchmine            MatchMine   
88         /organization/mediabong            MEDIABONG   
97   /organization/newscorporation            News Corp   
102          /organization/nokeena     Ankeena Networks   
104          /organization/nokeena     Ankeena Networks   
113           /organization/plumtv               PlumTV   

                      homepage_url                         category_list_x  \
7         http://alldefdigital.com                                   Media   
31        http://www.chefsfeed.com     Media|Mobile|Restaurants|Technology   
61   http://www.huffingtonpost.com                   Media|News|Publishing   
62   http://www.huffingtonpost.com                   Media|News|Publishing   
85            http://matchmine.com  Media|News|Reviews and Recommendations   
88        http://www.mediabong.com             Media|Semantic Search|Video   
97         http://www.newscorp.com                   Media|News|Publishing   
102         http://www.ankeena.com                          Media|Software   
104         http://www.ankeena.com                          Media|Software   
113          http://www.plumtv.com            Media|Television|Web Hosting   

        status country_code state_code         region           city  \
7    operating          USA         CA    Los Angeles    Los Angeles   
31   operating          USA         CA    SF Bay Area  San Francisco   
61    acquired          USA         NY  New York City       New York   
62    acquired          USA         NY  New York City       New York   
85      closed          USA         MA         Boston        Needham   
88   operating          USA         NY  New York City       New York   
97         ipo          USA         NY  New York City       New York   
102   acquired          USA         CA    SF Bay Area    Sa

In [35]:
# Total number of investments  and Total amount of investment(in USD) for USA

print(D1['main_sector'].count())

print(D1['raised_amount_usd'].sum())


12063
107757097294.0


In [36]:
# Finding the top 3 sectors based on number of investments for USA

pd.DataFrame(D1.groupby('main_sector').total_number_of_investment.mean()).sort_values(by='total_number_of_investment', ascending=False)


# FOR USA:-
# We can see that 'Others' have highest total_number_of_investment 
# followed by 'Social, Finance, Analytics, Advertising' and 'Cleantech / Semiconductors'

total_number_of_investment
main_sector                                                        
Others                                                         2950
Social, Finance, Analytics, Advertising                        2714
Cleantech / Semiconductors                                     2350
News, Search and Messaging                                     1583
Health                                                          909
Manufacturing                                                   799
Entertainment                                                   591
Automotive & Sports                                             167

In [37]:
#  For the top sector (Others) count-wise (point 3), which company received the highest investment?

D1.loc[(D1['main_sector']=='Others')].groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head(10)

permalink
/organization/virtustream           64300000.00
/organization/capella               54968051.00
/organization/airtight-networks     54201907.00
/organization/decarta               52100000.00
/organization/black-duck-software   51000000.00
/organization/approva               50100000.00
/organization/five9                 49600000.00
/organization/bit9                  48433533.00
/organization/aryaka-networks       48166500.00
/organization/bti-systems           48000000.00
Name: raised_amount_usd, dtype: float64

In [55]:
# Finding the name of the company which received the highest investment in Other sector

D1.loc[(D1['permalink']=='/organization/virtustream')].name.head(1)

40935    Virtustream
Name: name, dtype: object

In [39]:
#  For the second best sector count-wise (point 3), which company received the highest investment?

D1.loc[(D1['main_sector']=='Social, Finance, Analytics, Advertising')].groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head(10)

permalink
/organization/shotspotter    67933006.00
/organization/demandbase     63000000.00
/organization/intacct        61800000.00
/organization/netbase        60600000.00
/organization/lotame         59700000.00
/organization/firstrain      58344731.00
/organization/choicestream   58300000.00
/organization/optier         55600000.00
/organization/damballa       55000000.00
/organization/zoove          54800000.00
Name: raised_amount_usd, dtype: float64

In [40]:
# Finding the name of the company which received the highest investment 
# in 'Social, Finance, Analytics, Advertising' sector

D1.loc[(D1['permalink']=='/organization/shotspotter')].name.head(1)

16803    SST Inc. (Formerly ShotSpotter)
Name: name, dtype: object

-  ### Doing for country_code == GBR
     -  ### United Kingdom

In [41]:
# inspecting the created dataframe for GBR

D2.info()

D2.head(10)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 621 entries, 1 to 88247
Data columns (total 20 columns):
company_permalink             621 non-null object
funding_round_permalink       621 non-null object
funding_round_type            621 non-null object
funding_round_code            385 non-null object
funded_at                     621 non-null object
raised_amount_usd             621 non-null float64
permalink                     621 non-null object
name                          621 non-null object
homepage_url                  592 non-null object
category_list_x               621 non-null object
status                        621 non-null object
country_code                  621 non-null object
state_code                    607 non-null object
region                        592 non-null object
city                          592 non-null object
founded_at                    498 non-null object
primary_sector                621 non-null object
main_sector                   621 non-null

company_permalink  \
1                          /organization/90min   
2                          /organization/90min   
225  /organization/common-interest-communities   
257                /organization/geospock-ltd-   
258                /organization/geospock-ltd-   
367                  /organization/tao-group-2   
407                    /organization/workangel   
548                        /organization/azimo   
560             /organization/basekit-platform   
561             /organization/basekit-platform   

                             funding_round_permalink funding_round_type  \
1    /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
2    /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
225  /funding-round/8195587cbd5e51af7514ee92ef4ba6ba            venture   
257  /funding-round/cf3fe3b7c86186b9f478d0ea37613f7a            venture   
258  /funding-round/e5e4ef4ebae63fc36ef0cd57dd20ff1c            venture   
367  /funding-round/ad088f1deeda09f3338adfc324e32dab            venture   
407  /funding-round/3ff84c41cfa24575bd6ea60b78f580a6            venture   
548  /funding-round/5eb768935cf9c60b402944b0f476baae            venture   
560  /funding-round/8252cd70860ec66a1c7d13b6a2519dc6            venture   
561  /funding-round/9ceb098a822fb971db490b23c1067336            venture   

    funding_round_code   funded_at  raised_amount_usd  \
1                  NaN  06-10-2015        15000000.00   
2                  NaN  07-05-2013         5800000.00   
225                NaN  09-07-2014        10000000.00   
257                NaN  01-10-2014         5460000.00   
258                  A  05-10-2015         5400000.00   
367                NaN  06-05-2004         7000000.00   
407                  A  19-01-2015         5000000.00   
548                  A  11-03-2014        10000000.00   
560                NaN  13-05-2014         7000000.00   
561                NaN  03-02-2014         6818631.00   

                                     permalink                         name  \
1                          /organization/90min                        90min   
2                          /organization/90min                        90min   
225  /organization/common-interest-communities  Common Interest Communities   
257                /organization/geospock-ltd-                GeoSpock Ltd.   
258                /organization/geospock-ltd-                GeoSpock Ltd.   
367                  /organization/tao-group-2                    Tao Group   
407                    /organization/workangel                    WorkAngel   
548                        /organization/azimo                        Azimo   
560             /organization/basekit-platform                      BaseKit   
561             /organization/basekit-platform                      BaseKit   

                              homepage_url  \
1                     http://www.90min.com   
2                     http://www.90min.com   
225  http://commoninterestcommunities.com/   
257                http://www.geospock.com   
258                http://www.geospock.com   
367                  http://tao-group.com/   
407                  http://workangel.com/   
548                       http://azimo.com   
560                 http://www.basekit.com   
561                 http://www.basekit.com   

                                       category_list_x     status  \
1                  Media|News|Publishing|Soccer|Sports  operating   
2                  Media|News|Publishing|Soccer|Sports  operating   
225   Application Platforms|Internet|Software|Startups  operating   
257          Application Platforms|Databases|Real Time  operating   
258          Application Platforms|Databases|Real Time  operating   
367         Application Platforms|Consumer Electronics  operating   
407  Application Platforms|Employer Benefits Progra...  operating   
548  Curated Web|Finance Technology|Financial Servi...  operating   
560                    Cur

In [42]:
# Total number of investments  and Total amount of investment(in USD) for GBR

print(D2['main_sector'].count())

print(D2['raised_amount_usd'].sum())

621
5379078691.0


In [43]:
# Finding the top 3 sectors based on number of investments for GBR

pd.DataFrame(D2.groupby('main_sector').total_number_of_investment.mean()).sort_values(by='total_number_of_investment', ascending=False)


# FOR GBR (United Kingdom)
# We can see that 'Others' have highest total_number_of_investment 
# followed by 'Social, Finance, Analytics, Advertising' and 'Cleantech / Semiconductors'

total_number_of_investment
main_sector                                                        
Others                                                          147
Social, Finance, Analytics, Advertising                         133
Cleantech / Semiconductors                                      130
News, Search and Messaging                                       73
Entertainment                                                    56
Manufacturing                                                    42
Health                                                           24
Automotive & Sports                                              16

In [44]:
#  For the top sector(Others) count-wise (point 3), which company received the highest investment?

D2.loc[(D2['main_sector']=='Others')].groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head(10)

permalink
/organization/electric-cloud            37000000.00
/organization/sensage                   36250000.00
/organization/enigmatic                 32500000.00
/organization/silverrail-technologies   29000000.00
/organization/opencloud                 27972766.00
/organization/myoptique-group           27694613.00
/organization/notonthehighstreet        26900000.00
/organization/acs-clothing              26239014.00
/organization/tribold                   26000000.00
/organization/ip-access                 25000000.00
Name: raised_amount_usd, dtype: float64

In [45]:
# Finding the name of the company which received the highest investment in Others sector

D2.loc[(D2['permalink']=='/organization/electric-cloud')].name.head(1)

34333    Electric Cloud
Name: name, dtype: object

In [46]:
#  For the second best sector count-wise (point 3), which company received the highest investment?

D2.loc[(D2['main_sector']=='Social, Finance, Analytics, Advertising')].groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head(10)

permalink
/organization/celltick-technologies   37500000.00
/organization/mythings                34000000.00
/organization/zopa                    32900000.00
/organization/imagini                 28550000.00
/organization/marketinvoice           25553007.00
/organization/sumup                   24293649.00
/organization/amplience               24200000.00
/organization/garlik                  22350000.00
/organization/sportpursuit            22024487.00
/organization/victor                  21500000.00
Name: raised_amount_usd, dtype: float64

In [47]:
# Finding the name of the company which received the highest investment 
# in 'Social, Finance, Analytics, Advertising' sector

D2.loc[(D2['permalink']=='/organization/celltick-technologies')].name.head(1)

43677    Celltick Technologies
Name: name, dtype: object

-  ### Doing for country_code == IND
      -  ### India

In [48]:
# inspecting the created dataframe for India

D3.info()

D3.head(10)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 328 entries, 0 to 88139
Data columns (total 20 columns):
company_permalink             328 non-null object
funding_round_permalink       328 non-null object
funding_round_type            328 non-null object
funding_round_code            228 non-null object
funded_at                     328 non-null object
raised_amount_usd             328 non-null float64
permalink                     328 non-null object
name                          328 non-null object
homepage_url                  325 non-null object
category_list_x               328 non-null object
status                        328 non-null object
country_code                  328 non-null object
state_code                    327 non-null object
region                        327 non-null object
city                          327 non-null object
founded_at                    282 non-null object
primary_sector                328 non-null object
main_sector                   328 non-null

company_permalink  \
0                /organization/-fame   
550            /organization/babajob   
593   /organization/bharat-matrimony   
640      /organization/bluestone-com   
642      /organization/bluestone-com   
1205   /organization/fourinteractive   
1657        /organization/localbanya   
1664          /organization/localoye   
1818          /organization/mobikwik   
1832   /organization/money-on-mobile   

                              funding_round_permalink funding_round_type  \
0     /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
550   /funding-round/b72eaac5ea12ac0f50573ac3d6d46b8d            venture   
593   /funding-round/e37673bc7b0f1dfd3782f8f7abdb9ec8            venture   
640   /funding-round/452a7fc1f34df2d3dcda4e28234bc671            venture   
642   /funding-round/f5b252d6442ce231bb01586ca1821f63            venture   
1205  /funding-round/a0ef5f6e0c69a9060c1649c076999336            venture   
1657  /funding-round/087fbb346606a864c03199ec3189e67b            venture   
1664  /funding-round/b56bf538e5e50ae038359a6334862a5a            venture   
1818  /funding-round/6cb899c717aab8ff314ca4b257124377            venture   
1832  /funding-round/fe6c80376b0e82118d2716049a1f411c            venture   

     funding_round_code   funded_at  raised_amount_usd  \
0                     B  05-01-2015        10000000.00   
550                   B  28-04-2015        10000000.00   
593                   B  05-02-2008        11750000.00   
640                   A  24-01-2012         5000000.00   
642                   B  18-03-2014        10000000.00   
1205                  B  01-09-2007        10000000.00   
1657                  A  09-01-2014         5000000.00   
1664                  A  14-04-2015         5000000.00   
1818                  A  01-01-2013         5000000.00   
1832                NaN  16-09-2013        10000000.00   

                           permalink              name  \
0                /organization/-fame             #fame   
550            /organization/babajob           Babajob   
593   /organization/bharat-matrimony  Bharat Matrimony   
640      /organization/bluestone-com     Bluestone.com   
642      /organization/bluestone-com     Bluestone.com   
1205   /organization/fourinteractive  Four Interactive   
1657        /organization/localbanya        LocalBanya   
1664          /organization/localoye          LocalOye   
1818          /organization/mobikwik          MobiKwik   
1832   /organization/money-on-mobile   Money On Mobile   

                        homepage_url  \
0                 http://livfame.com   
550           http://www.babajob.com   
593   http://www.bharatmatrimony.com   
640             http://bluestone.com   
642             http://bluestone.com   
1205          http://www.fourint.com   
1657           http://localbanya.com   
1664             http://localoye.com   
1818         http://www.mobikwik.com   
1832  http://www.money-on-mobile.net   

                                        category_list_x     status  \
0                                                 Media  operating   
550   Curated Web|Information Technology|Services|St...  operating   
593                            Curated Web|Match-Making  operating   
640                                         Curated Web  operating   
642                                         Curated Web  operating   
1205                                        Curated Web  operating   
1657                                        Curated Web  operating   
1664                                        Curated Web  operating   
1818  Curated Web|Internet|Mobile|Mobile Payments|Pa...  operating   
1832                                 Curated Web|Mobile  operating   

     country_code state_code     region       city  founded_at primary_sector  \
0             IND         16     Mumbai     Mumbai         NaN          media   
550           IND         19  Bangalore  Bangalore  27-08-2007    curated web   
593     

In [49]:
# Total number of investments  and Total amount of investment(in USD) for IND

print(D3['main_sector'].count())

print(D3['raised_amount_usd'].sum())

328
2949543602.0


In [50]:
# Finding the top 3 sectors based on number of investments for IND

pd.DataFrame(D3.groupby('main_sector').total_number_of_investment.mean()).sort_values(by='total_number_of_investment', ascending=False)


# FOR IND
# We can see that 'Others' have highest total_number_of_investment 
# followed by 'Social, Finance, Analytics, Advertising' and 'News, Search and Messaging'

total_number_of_investment
main_sector                                                        
Others                                                          110
Social, Finance, Analytics, Advertising                          60
News, Search and Messaging                                       52
Entertainment                                                    33
Manufacturing                                                    21
Cleantech / Semiconductors                                       20
Health                                                           19
Automotive & Sports                                              13

In [51]:
#  For the top sector(Others) count-wise (point 3), which company received the highest investment?

D3.loc[(D3['main_sector']=='Others')].groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head(10)

permalink
/organization/firstcry-com                                                             39000000.00
/organization/myntra                                                                   38000000.00
/organization/commonfloor                                                              32900000.00
/organization/pepperfry-com                                                            28000000.00
/organization/itzcash-card-ltd                                                         25000000.00
/organization/nxtgen-data-center-cloud-services                                        22300000.00
/organization/maharana-infrastructure-and-professional-services-private-limited-mips   21600000.00
/organization/caratlane                                                                21000000.00
/organization/comat-technologies                                                       21000000.00
/organization/limeroad                                                                 20000000.00


In [52]:
# Finding the name of the company which received the highest investment in Others sector

D3.loc[(D3['permalink']=='/organization/firstcry-com')].name.head(1)

21181    FirstCry.com
Name: name, dtype: object

In [53]:
#  For the second best sector count-wise (point 3), which company received the highest investment?

D3.loc[(D3['main_sector']=='Social, Finance, Analytics, Advertising')].groupby('permalink')['raised_amount_usd'].sum().sort_values(ascending=False).head(10)

permalink
/organization/manthan-systems                                50700000.00
/organization/komli-media                                    28000000.00
/organization/shopclues-com                                  25000000.00
/organization/intarvo                                        21900000.00
/organization/grameen-financial-services                     21556050.00
/organization/bankbazaar                                     19000000.00
/organization/microland                                      18300000.00
/organization/eka-software-solutions                         16000000.00
/organization/qyuki                                          15400000.00
/organization/financial-information-network-operations-pvt   15000000.00
Name: raised_amount_usd, dtype: float64

In [54]:
# Finding the name of the company which received the highest investment 
# in 'Social, Finance, Analytics, Advertising' sector

D3.loc[(D3['permalink']=='/organization/manthan-systems')].name.head(1)

16003    Manthan Systems
Name: name, dtype: object